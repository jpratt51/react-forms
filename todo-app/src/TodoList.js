import React, { useState } from "react";
import { v4 as uuid } from "uuid";
import NewTodoForm from "./NewTodoForm";

const TodoList = () => {
    const INITIAL_STATE = [
        { id: uuid(), todo: "mow yard" },
        { id: uuid(), todo: "dishes" },
    ];
    const [todos, setTodos] = useState(INITIAL_STATE);
    const addTodo = (newTodo) => {
        setTodos((todos) => [...todos, { ...newTodo, id: uuid() }]);
    };
    const deleteTodo = (id) => {
        setTodos((todos) => todos.filter((box) => box.id !== id));
    };
    return (
        <div>
            <h3>Todos</h3>
            <NewTodoForm addTodo={addTodo} />
            {todos.map(({ id, todo }) => (
                <div key={id}>
                    <div id={id}>{todo}</div>
                    <button onClick={() => deleteTodo(id)}>X</button>
                </div>
            ))}
        </div>
    );
};

export default TodoList;
