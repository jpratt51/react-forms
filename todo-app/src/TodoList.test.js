import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TodoList from "./TodoList";

it("renders without crashing", function () {
    render(<TodoList />);
});

it("matches snapshot", function () {
    const { asFragment } = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot();
});

it("should add new todo", function () {
    const { getByLabelText, queryByText, queryByTestId } = render(<TodoList />);
    const input = getByLabelText("Todo Item");
    const btn = queryByText("Add Todo");
    expect(queryByText("cook dinner")).not.toBeInTheDocument();
    fireEvent.change(input, { target: { value: "cook dinner" } });
    fireEvent.click(btn);
    expect(queryByText("cook dinner")).toBeInTheDocument();
});

it("should delete todo", function () {
    const { queryByTestId } = render(<TodoList />);
    const btn = queryByTestId("mow yard");
    expect(queryByTestId("mow yard")).toBeInTheDocument();
    fireEvent.click(btn);
    expect(queryByTestId("mow yard")).not.toBeInTheDocument();
});
