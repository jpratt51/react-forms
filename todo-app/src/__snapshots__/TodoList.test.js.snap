// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      Todos
    </h3>
    <form>
      <label
        for="todoItem"
      >
        Todo Item 
      </label>
      <input
        id="todoItem"
        name="todoItem"
        placeholder="Todo Item"
        type="text"
        value=""
      />
      <button>
        Add Todo
      </button>
    </form>
    <div>
      <div
        id="71ca6571-2fb3-4f20-a8a4-0cd2fd86f811"
      >
        mow yard
      </div>
      <button
        data-testid="mow yard"
      >
        X
      </button>
    </div>
    <div>
      <div
        id="0f45917d-6da0-4f45-87a8-58441ea6947c"
      >
        dishes
      </div>
      <button
        data-testid="dishes"
      >
        X
      </button>
    </div>
  </div>
</DocumentFragment>
`;
